app/src/org/commcare/services/WiFiDirectBroadcastReceiver.java

```diff
@@ -76,19 +83,25 @@ public void onReceive(Context context, Intent intent) {
                 return;
             }
 
-            NetworkInfo networkInfo = intent
-                    .getParcelableExtra(WifiP2pManager.EXTRA_NETWORK_INFO);
+            NetworkInfo networkInfo;
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
+                networkInfo = intent
+                        .getParcelableExtra(WifiP2pManager.EXTRA_NETWORK_INFO);
+            } else {
+                //EXTRA_NETWORK_INFO is removed from Android API 28
+                networkInfo = connectivityManager.getActiveNetworkInfo();
```

Comment:
@rristovic  I looked a bit into this and it seems like `EXTRA_NETWORK_INFO` got removed for the broadcast `NETWORK_STATE_CHANGED_ACTION` and not for `WifiP2pManager.WIFI_P2P_CONNECTION_CHANGED_ACTION`. Also here `connectivityManager` will return whether the phone is connected to internet while the `networkInfo` earlier was supposed to convey the info regarding  wifi P2P connection for ex. if we have a P2P connection between 2 devices.